src/icalendar/cal/todo.py

```diff
@@ -215,11 +215,143 @@ def end(self, end: date | datetime | None):
     def duration(self) -> timedelta:
         """The duration of the VTODO.
 
-        This duration is calculated from the start and end of the Todo.
-        You cannot set the duration as it is unclear what happens to start and end.
+        Returns the DURATION property if set, otherwise calculated from start and end.
+        You can set the duration to automatically adjust the end time while keeping start locked.
         """
+        # First check if DURATION property is explicitly set
+        if "DURATION" in self:
+            # For incomplete todos with only DURATION, we need to be able to compute end
+            # which requires either DTSTART or DUE
+            if self.DTSTART is None and self.DUE is None:
+                # This matches the behavior expected by the test
+                raise IncompleteComponent("No DUE or DURATION+DTSTART given.")
+            return self["DURATION"].dt
+
+        # Fall back to calculated duration from start and end
         return self.end - self.start
 
+    @duration.setter
+    def duration(self, value: timedelta):
+        """Set the duration of the todo.
+
+        Setting the duration will:
+        1. Keep the start time locked (unchanged)
+        2. Adjust the end time to start + duration
+        3. Remove any existing DUE property
+        4. Set the DURATION property
+        """
+        if not isinstance(value, timedelta):
+            raise TypeError(f"Use timedelta, not {type(value).__name__}.")
+
+        # Use the set_duration method with default start-locked behavior
+        self.set_duration(value, locked="start")
+
+    def set_duration(self, duration: timedelta | None, locked: str = "start"):
+        """Set the duration with explicit locking behavior.
```

Comment:
Please also describe what you are locking so that people know.

I would like this.
```suggestion
        """Set the duration and keep either start or end.
```